enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -0,0 +1,867 @@
+spec:
+  installNamespace: open-cluster-management-agent-addon
+Status:
+…
+  lastVersion: v1
+  currentVersion: v2
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: install completed with no errors.
+    reason: Succeed
+    status: "False"
+    type: Progressing
+```
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata:
+  labels:
+    open-cluster-management.io/addon-name: application-manager
+    open-cluster-management.io/addon-version: v2
+  generation: 2
+  name: addon-application-manager-deploy
+  namespace: cluster2
+…
+status:
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:47:30Z"
+    message: All resources are available
+    observedGeneration: 2
+    reason: ResourcesAvailable
+    status: "True"
+    type: Available
+```
+
+#### Canary upgrade addon to specific version.
+Users can modify the installStrategy to achieve canary upgrade.
+
+1. Define the installedVerison explicitly in global-placement before upgrade.
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: global-placement
+      namespace: default
+      addonTemplate:
+        desiredVersion: v1
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: global-config
+status:
+  supportedVersions:
+  - v1
+  defaultVersion: v1
+```
+
+2. Upgrade the ClusterManagementAddOn and addon manager.
+
+After the addon manager is upgraded, the supportedVersions and defaultVersion is also updated by addon manager.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: global-placement
+      namespace: default
+      addonTemplate:
+        desiredVersion: v1
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: global-config
+status:
+  supportedVersions:
+  - v1
+  - v2
+  defaultVersion: v2
+```
+
+3. Add new placement to select a group of clusters to upgrade to the new version addon.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: global-placement
+      namespace: default
+      addonTemplate:
+        desiredVersion: v1
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: global-config
+    - name: canary-placement
+      namespace: default
+      addonTemplate:
+        desiredVersion: v2
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: canary-config
```

Comment:
I'm going to experiment with kustomize, but here's where my head is at.  If we consider that configuration changes are going to be treated the same and that often we even want to choose both, I'm wondering if using a gitops flow and having desiredVersion on the addondeploymentconfigs type and then using the "add-sha-suffix" option in kustomize and then substituting that in this name.  

That might conveniently solve version updates (with arbitrary skew), configuration changes, gitops flows, and gitops reverts, and manual management.